### PR TITLE
add support for Microchip SmartFusion2 family FPGA

### PR DESCRIPTION
**Pull request by whik**
add support for Microchip SmartFusion2 family FPGA

### CODE REVIEW

File: board/Microchip_M2S010/BSP/Inc/config.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * 2020-08-06     whik         first version
+ */
+#ifndef __CONFIG_H__
+#define __CONFIG_H__
+
+#include "mss_gpio.h"
+#include "mss_uart.h"
+
+#include <rthw.h>
+#include <rtthread.h>
```
Comment (Supowang1989) [inline board/Microchip_M2S010/BSP/Inc/config.h:17]: Thanks for your pull request，please remove the code form RT-Thread，Then we can merge this.

File: board/Microchip_M2S010/BSP/Src/board.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author       Notes
+ * 2020-08-06     whik         first version
+ */
+ 
+#include <stdint.h>
+#include <rthw.h>
+#include <rtthread.h>
```
Comment (Supowang1989) [inline board/Microchip_M2S010/BSP/Src/board.c:13]: Thanks for your pull request，please remove the code form RT-Thread，Then we can merge this.

**Comment by Supowang1989** [comment]
Thanks for your pull request，please remove the code form RT-Thread，Then we can merge it.

File: board/Microchip_M2S010/BSP/Src/board.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    
+    _SYSTICK_LOAD = ticks - 1; 
+    _SYSTICK_PRI = 0xFF;
+    _SYSTICK_VAL  = 0;
+    _SYSTICK_CTRL = 0x07;  
+    
+    return 0;
+}
+
+#if defined(RT_USING_USER_MAIN) && defined(RT_USING_HEAP)
```
Comment (Supowang1989) [inline board/Microchip_M2S010/BSP/Src/board.c:40]: Thanks for your pull request，please remove the code form RT-Thread，Then we can merge this.

File: board/Microchip_M2S010/BSP/Src/board.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    /* Call components board initial (use INIT_BOARD_EXPORT()) */
+#ifdef RT_USING_COMPONENTS_INIT
+    rt_components_board_init();
+#endif
+
+#ifdef RT_USING_CONSOLE
+    rt_console_set_device(RT_CONSOLE_DEVICE_NAME);
+#endif
+    
+#if defined(RT_USING_USER_MAIN) && defined(RT_USING_HEAP)
```
Comment (Supowang1989) [inline board/Microchip_M2S010/BSP/Src/board.c:72]: Thanks for your pull request，please remove the code form RT-Thread，Then we can merge this.

Comment (whik) [reply board/Microchip_M2S010/BSP/Src/board.c]: haha, These files are not used.

**Comment by Supowang1989** [comment]
Thanks

**Comment by Supowang1989** [comment]
maybe we can add WeChat to keep in touch， my wecat：supowang

**Pull request merged**